federation/pkg/federation-controller/sync/controller.go

```diff
@@ -444,10 +445,38 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		return statusError
 	}
 
-	operations, err := operationsAccessor(adapter, selectedClusters, unselectedClusters, obj)
+	var schedulingInfo *federatedtypes.SchedulingInfo
+	if adapter.IsSchedulingAdapter() {
+		schedulingAdapter, ok := adapter.(federatedtypes.SchedulingAdapter)
+		if !ok {
+			glog.Fatalf("Adapter for kind %q does not properly implement SchedulingAdapter.", kind)
+			return statusError
+		}
+		schedulingInfo, err = schedulingAdapter.GetSchedule(obj, key, selectedClusters, informer)
```

Comment:
Please ensure that, the cluster objects are queried and passed as one set here (probably as a map), indicating which cluster has the object, and which does not (one way is use a map of same size as current available clusters with nil values identifying absence of object in that cluster).
Other option I could think of is;  don't query the objects here at all; just pass the informer here, and let this interface or any other one in the flow later return the existing set vs scheduled set of objects (or only scheduled set of objects indicating, in which cluster to update, into which cluster to delete and which cluster to create ... (Or else I can't think of how exactly to do it ).
Probably u can think of on these lines:
say in the rs controller schedule, as a result of schedule, the local cluster rs can be deleted from some clusters also
so there has to be some way for this reconcile to be able to finalise the result comparing the original set of object vs new set of objects including all three conditions:
- if not exist in a cluster in existing set,  schedule says create it.
- if exist in a cluster in existing set, schedule says update it.
- if exist in a cluster in existing set, schedule says delete it.

Also the above to be done without race conditions (means objects or set of cluster objects queried at one location to come up with the schedule result)